PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx

```diff
@@ -220,25 +222,32 @@ struct HfTaskMcEfficiencyToXiPi {
       }
 
       // exclude cases with undesired decays
-      if (mcParticle.daughtersIds().size() != 2) {
+      int counterCharmDau = 0;
+      for (auto& dauCharm : mcParticle.template daughters_as<T2>()) {
+        if (dauCharm.getProcess() == TMCProcess::kPDecay || dauCharm.getProcess() == TMCProcess::kPPrimary) {
+          counterCharmDau++;
+        }
+      }
+      if (counterCharmDau != 2) {
         LOGP(fatal, "Invalid numbers of daughters for charm baryon {}: {}", mcParticle.globalIndex(), mcParticle.daughtersIds().size());
       }
-      auto cascId = mcParticle.daughtersIds()[0];
-      auto pionId = mcParticle.daughtersIds()[1];
+      int cascId = -999;
+      int pionId = -999;
+      for (auto& dauCharm : mcParticle.template daughters_as<T2>()) {
+        if (std::abs(dauCharm.pdgCode()) == kXiMinus && (dauCharm.getProcess() == TMCProcess::kPDecay || dauCharm.getProcess() == TMCProcess::kPPrimary)) {
+          cascId = dauCharm.globalIndex();
+        } else if (std::abs(dauCharm.pdgCode()) == kPiPlus && (dauCharm.getProcess() == TMCProcess::kPDecay || dauCharm.getProcess() == TMCProcess::kPPrimary)) {
+          pionId = dauCharm.globalIndex();
+        } else {
+          continue;
+        }
+      }
```

Comment:
This part could be merged in the loop of L226. Moreover, I think `continue` is doing nothing here since it should just go to the next for-loop iteration.